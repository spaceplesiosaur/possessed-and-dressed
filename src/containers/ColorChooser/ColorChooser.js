import React, { Component } from 'react'
import './ColorChooser.scss'
import { connect } from 'react-redux'

export default class ColorChooser extends Component {
  constructor({colorName, hues, borderColor}) {
    super()
    this.state = {
      listDisplay: 'colorList-hues',
    }
  }

  hydrateColorList = () => {
    this.props.hues.map(hue => {
      return <li className="colorList-box" background-color={hue}></li>
    })
  }
  render() {
    return (
      <div className="colorChooser-colorDropdown-widget">
        <h4 className="colorList-box" border={`3px solid ${this.props.borderColor}`}>{this.props.colorName}</h4>
        <ul className={this.state.listDisplay}>
          {this.hydrateColorList()}
        </ul>
      </div>
    )
  }
}
