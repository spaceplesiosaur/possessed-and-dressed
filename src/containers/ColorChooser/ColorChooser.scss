.hidden {
  display: none;
}
.colorChooser-colorDropdown-widget {
  display: flex;
  flex-direction: column;
  height: 400px;
}

.colorList-box {
  width: 9.5em;
  height: 2.5em;
  margin-left: .75em;
  margin-right: .75em;
  background-color: white;
}

.colorList-hues {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  list-style-type: none;
}
