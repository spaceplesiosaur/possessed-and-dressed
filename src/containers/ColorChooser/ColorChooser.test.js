import React from 'react';
import { ColorChooser } from './ColorChooser';
import { shallow } from 'enzyme';

describe('ColorChooser', () => {
  let wrapper;
  let mockHues;
  let mockChosenHost;
  let mockChosenColor;

  beforeEach(() => {
    mockChosenHost = {
        id: 1,
        name: "Blucifer",
        picture: "https://i.imgur.com/I5MqMUV.jpg",
        season: 9
    }

    mockChosenColor = {
        id: 41,
        name: "watermelon",
        hex_code: "#EE3F4B",
        category: 34
    }

    mockHues = [{
        id: 45,
        name: "astros",
        hex_code: "#ff6100",
        category: 35
    },
    {
        id: 44,
        name: "burnt orange",
        hex_code: "#e5732f",
        category: 35
    },
    {
        id: 43,
        name: "tangelo",
        hex_code: "#FF9500",
        category: 35
    }]

    wrapper = shallow(<ColorChooser
      colorName={"red"}
      hues={mockHues}
      borderColor={"#F87E97"}
      chosenHost={mockChosenHost}
      chosenColor={""}
      />)
  })

  it('should render with the correct data', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with the correct data when there is a chosen color', () => {
    const wrapper = shallow(<ColorChooser
      colorName={"red"}
      hues={mockHues}
      borderColor={"#F87E97"}
      chosenHost={mockChosenHost}
      chosenColor={mockChosenColor}
      />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should toggle the state to open and closed', () => {
    wrapper.instance().setState({open: false})
    wrapper.instance().toggleListDisplay()
    expect(wrapper.instance().state.open).toEqual(true)

    wrapper.instance().setState({open: true})
    wrapper.instance().toggleListDisplay()
    expect(wrapper.instance().state.open).toEqual(false)
  })
})
