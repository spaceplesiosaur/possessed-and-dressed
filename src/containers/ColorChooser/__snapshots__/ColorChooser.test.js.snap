// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorChooser should render with the correct data 1`] = `
<div
  className="colorChooser-colorDropdown-widget"
>
  <h4
    className="colorList-box"
    onClick={[Function]}
    style={
      Object {
        "border": "10px solid #F87E97",
      }
    }
  >
    red
  </h4>
  <ul
    className="hidden"
  >
    <li
      className="colorList-box"
      id="astros"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#ff6100",
        }
      }
    />
    <li
      className="colorList-box"
      id="burnt orange"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e5732f",
        }
      }
    />
    <li
      className="colorList-box"
      id="tangelo"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FF9500",
        }
      }
    />
  </ul>
</div>
`;

exports[`ColorChooser should render with the correct data when there is a chosen color 1`] = `
<Redirect
  to="/hosts/1/watermelon"
/>
`;
